internal/ir/xds.go

```diff
@@ -444,29 +444,27 @@ type HTTPRoute struct {
 	Destination *RouteDestination `json:"destination,omitempty" yaml:"destination,omitempty"`
 	// Rewrite to be changed for this route.
 	URLRewrite *URLRewrite `json:"urlRewrite,omitempty" yaml:"urlRewrite,omitempty"`
+	// ExtensionRefs holds unstructured resources that were introduced by an extension and used on the HTTPRoute as extensionRef filters
+	ExtensionRefs []*UnstructuredRef `json:"extensionRefs,omitempty" yaml:"extensionRefs,omitempty"`
+
+	BackendTrafficPolicy `json:",inline" yaml:",inline"`
```

Comment:
Using named fields instead of embedded fields is preferable in this tree structure. Embedding is better suited for scenarios involving unioned interfaces or inheritance. Since child nodes may share methods (Any, Validate, Set, etc.), named fields help avoid potential naming collisions and express a clear compostion/tree structure.